packages/nvidia_nat_maf/src/nat/plugins/maf/tool_wrapper.py

```
@@ -0,0 +1,149 @@
+    if origin in (Union, types.UnionType):
+        # Pick the first type that isn’t NoneType
+        for arg in get_args(t):
+            if arg is not type(None):
+                return arg
+
+        return t  # fallback
+    return t
+
+
+@register_tool_wrapper(wrapper_type=LLMFrameworkEnum.MICROSOFT_AGENT_FRAMEWORK) # 1. Changed framework type
+def microsoft_agent_framework_tool_wrapper(name: str, fn: Function, builder: Builder):
```

Ruff (0.14.7)

Unused function argument: `builder`

(ARG001)